### Note
* Instructions have been included for each segment. You do not have to follow them exactly, but they are included to help you think through the steps.

In [279]:
# Dependencies and Setup
import pandas as pd

# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)
purchase_data


Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male      101   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price  
0    Extraction, Quickblade Of Trembling Hands   3.53  
1                            Frenzied Scimitar   1.56  
2                                 Final Critic   4.88  
3                                  Blindscythe   3.27  
4                                         Fury   1.44  
..                                         ...    ...  
775                                       Wolf   3.54  
776                           Exiled Doomblade   1.63  
777      Celeste, Incarnation of the Corrupted   3.46  
778                               Final Critic   4.19  
779                                       Dawn   4.60  

[780 rows x 7 columns]

## Player Count

* Display the total number of players


In [281]:
total=purchase_data["SN"].nunique()
total_players= {"total Players":[total]}
df=pd.DataFrame(total_players,index=[0])
df


total Players
0            576

## Purchasing Analysis (Total)

* Run basic calculations to obtain number of unique items, average price, etc.


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame


In [253]:
#number of unique items
purchase_data_groubby= purchase_data.groupby("Item ID")
unique_item=len(purchase_data_groubby["Item ID"].nunique())

unique_item

183

In [254]:
#average price(I could not round a single value)
average_price=purchase_data["Price"].mean()
#average_price="${:,.2f}".format(purchase_data["Price"].mean())
##purchase_data["Price"]=purchase_data["Price"].map("${:,.2f}".format)
average_price

3.050987179487176

In [255]:
#number of perchases
number_purchases=purchase_data["Item ID"].count()
number_purchases

780

In [256]:
#Total Revenue
Total_Revenue=purchase_data["Price"].sum()
Total_Revenue


2379.77

In [283]:
Purchasing_Analysis=pd.DataFrame({"number of Unique Items":unique_item,
                                 "Average Price":[average_price],
                                 "Number of Purchases":number_purchases,
                                  "Total Revenue":[Total_Revenue]},index=[0])

Purchasing_Analysis


number of Unique Items  Average Price  Number of Purchases  Total Revenue
0                     183       3.050987                  780        2379.77

In [258]:
Purchasing_Analysis["Average Price"]=Purchasing_Analysis["Average Price"].map("${:,.2f}".format)

Purchasing_Analysis["Total Revenue"]=Purchasing_Analysis["Total Revenue"].map("${:,.2f}".format)
Purchasing_Analysis

number of Unique Items Average Price  Number of Purchases Total Revenue
0                     183         $3.05                  780     $2,379.77

## Gender Demographics

* Percentage and Count of Male Players


* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [259]:
#purchase_data.groupby("Gender")


In [284]:
#purchase_data.groupby("Gender")

total_count=purchase_data.groupby(["Gender"]).nunique()
total_count_percentage=total_count["SN"]/total*100

total_count_percentage=total_count_percentage.map("{:,.2f}%".format)

Gender_Demographics=pd.DataFrame({"Total Count":total_count["SN"],
                           "Percentage of Players":total_count_percentage},
                                  index=["Male", "Female", "Other / Non-Disclosed"])

Gender_Demographics


Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male      101   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price  
0    Extraction, Quickblade Of Trembling Hands   3.53  
1                            Frenzied Scimitar   1.56  
2                                 Final Critic   4.88  
3                                  Blindscythe   3.27  
4                                         Fury   1.44  
..                                         ...    ...  
775                                       Wolf   3.54  
776                           Exiled Doomblade   1.63  
777      Celeste, Incarnation of the Corrupted   3.46  
778                               Final Critic   4.19  
779                                       Dawn   4.60  

[780 rows x 7 columns]

In [261]:
total_count

Purchase ID   SN  Age  Gender  Item ID  Item Name  \
Gender                                                                     
Female                         113   81   22       1       90         90   
Male                           652  484   39       1      182        178   
Other / Non-Disclosed           15   11    8       1       13         13   

                       Price  
Gender                        
Female                    79  
Male                     144  
Other / Non-Disclosed     12

In [263]:
#purchase_data=purchase_data.drop_duplicates(inplace=True)



#male_players=len(purchase_data["Gender"].loc[purchase_data["Gender"]=="Male"])
#female_players=len(purchase_data["Gender"].loc[purchase_data["Gender"]=="Female"])
#tother_players=len(purchase_data["Gender"])-male_players-female_players
#total_players=len(purchase_data["Gender"])
#Gender_Demographics=pd.DataFrame({"Total Count":[male_players,female_players,other_players],
#                                  "Percentage of Players":[male_players/total_players*100,female_players/total_players*100,other_players/total_players*100]                           
#    },index=["Male", "Female", "Other / Non-Disclosed"])
#Gender_Demographics


## Purchasing Analysis (Gender)

* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender




* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [286]:

total_count=purchase_data.groupby(["Gender"]).nunique()
total_count_percentage=total_count["SN"]/total*100

total_count_percentage=total_count_percentage.map("{:,.2f}%".format)

Gender_Demographics=pd.DataFrame({"Total Count":total_count["SN"],
                           "Percentage of Players":total_count_percentage},
                                  index=["Male", "Female", "Other / Non-Disclosed"])

Gender_Demographics


Total Count Percentage of Players
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

In [288]:
#purchase count
pd_group_gender=purchase_data.groupby(["Gender"])
total_count_2=pd_group_gender["SN"].count()

#average price
ave_count=pd_group_gender["Price"].mean()
ave_count=ave_count.map("{:,.2f}%".format)

#Total Purchase Value(T_P_V)
T_P_V=pd_group_gender["Price"].sum()

#Avg Total Purchase per Person(A_T_P_P_P)
A_T_P_P_P=T_P_V/Gender_Demographics["Total Count"]

#clear formating
T_P_V=T_P_V.map("${:,.2f}".format)
A_T_P_P_P=A_T_P_P_P.map("${:,.2f}%".format)

#Create a summary data frame to hold the results 
summery_gender=pd.DataFrame({"Total Count":total_count_2,
                           "Average Price":ave_count,
                        "Total Purchase Value":T_P_V,
                          "Avg Total Purchase Per Person": A_T_P_P_P 
                        })
summery_gender


Total Count Average Price Total Purchase Value  \
Gender                                                                  
Female                         113         3.20%              $361.94   
Male                           652         3.02%            $1,967.64   
Other / Non-Disclosed           15         3.35%               $50.19   

                      Avg Total Purchase Per Person  
Gender                                               
Female                                       $4.47%  
Male                                         $4.07%  
Other / Non-Disclosed                        $4.56%

## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


In [290]:
#Age Demographics
bins=[0,9,14,19,24,29,34,39,2000]
labels_group=["<10","10-14","15-19","20-24","25-29","30-34","35-39","40+"]
purchase_data["score summery"]=pd.cut(purchase_data["Age"],bins, labels=labels_group)

age_group=purchase_data.groupby(["score summery"])

#Calculate the numbers and percentages by age group(n_age_groups) and (p_age_groups)
n_age_groups=age_group["SN"].nunique()
p_age_groups=n_age_groups/total*100

# round the percentage column to two decimal points
p_age_groups=p_age_groups.map("{:,.2f}%".format)


#Create a summary data frame to hold the results 
Age_Demographics=pd.DataFrame({"Total Count":n_age_groups,
                           "Percentage of Players":p_age_groups,
                       
                        })
Age_Demographics



Total Count Percentage of Players
score summery                                   
<10                     17                 2.95%
10-14                   22                 3.82%
15-19                  107                18.58%
20-24                  258                44.79%
25-29                   77                13.37%
30-34                   52                 9.03%
35-39                   31                 5.38%
40+                     12                 2.08%

## Purchasing Analysis (Age)

* Bin the purchase_data data frame by age


* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [267]:
purchase_data["score summery"]=pd.cut(purchase_data["Age"],bins, labels=labels_group)
age_group=purchase_data.groupby(["score summery"])

In [375]:
#purchase count
total_count_age=age_group["Purchase ID"].nunique()

#Average Purchase Price(A_P_P_A)
A_P_P_A=age_group["Price"].mean()

#Total purchase value (abreviation:tpv)
tpv=total_count_age*A_P_P_A

#Avg Total Purchase per Person(abreviation:atpp),total count including repetetive customer(abreviation:tcirc)
tcirc=age_group["Purchase ID"].count()
atpp =tpv/Age_Demographics["Total Count"]

#give the displayed data cleaner formatting
A_P_P_A=A_P_P_A.map("${:,.2f}".format)
tpv=tpv.map("${:,.2f}".format)
atpp=atpp.map("${:,.2f}%".format)

#Create a summary data frame to hold the results
Purchasing_Analysis_Age=pd.DataFrame({"purchase count":total_count_age,
                                     "Average Purchase Price":A_P_P_A,
                                      "Total purchase value":tpv,
                                      "Avg Total Purchase per Person":atpp                                   
                                     
                                     })
                        
Purchasing_Analysis_Age




purchase count Average Purchase Price Total purchase value  \
score summery                                                               
<10                        23                  $3.35               $77.13   
10-14                      28                  $2.96               $82.78   
15-19                     136                  $3.04              $412.89   
20-24                     365                  $3.05            $1,114.06   
25-29                     101                  $2.90              $293.00   
30-34                      73                  $2.93              $214.00   
35-39                      41                  $3.60              $147.67   
40+                        13                  $2.94               $38.24   

              Avg Total Purchase per Person  
score summery                                
<10                                  $4.54%  
10-14                                $3.76%  
15-19                                $3.86%  
20-24                                $4.32%  
25-29                                $3.81%  
30-34                                $4.12%  
35-39                                $4.76%  
40+                                  $3.19%

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [396]:
#number of unique items(abreviation:nui)
#Group by Spendors ( "SN" )
nui=purchase_data.groupby(["SN"])
nui_count=nui["Purchase ID"].count()


#Average Purchase Price
nui_ave=nui["Price"].mean()


#Total Purchase Value
nui_sum=nui["Price"].sum()

#give the displayed data cleaner formatting
nui_ave=nui_ave.map("${:,.2f}".format)


#Create a summary data frame to hold the results
Top_Spenders=pd.DataFrame({"purchase count":nui_count,
                                     "Average Purchase Price":nui_ave,
                                      "Total purchase value":nui_sum,
                                     
                                     })
#Sort the total purchase value column in descending order

Top_Spenders=Top_Spenders.sort_values("Total purchase value",ascending=False)


#give the displayed data cleaner formatting
Top_Spenders["Total purchase value"]=nui_sum.map("${:,.2f}".format)


Top_Spenders

purchase count Average Purchase Price Total purchase value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10
...                     ...                    ...                  ...
Ililsasya43               1                  $1.02                $1.02
Irilis75                  1                  $1.02                $1.02
Aidai61                   1                  $1.01                $1.01
Chanirra79                1                  $1.01                $1.01
Alo38                     1                  $1.00                $1.00

[576 rows x 3 columns]

## Most Popular Items

* Retrieve the Item ID, Item Name, and Item Price columns


* Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value


* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [443]:
#Group by Item ID and Item Name(abbreviation:giiin)
giiin=purchase_data.groupby(["Item ID","Item Name"])

#Purchase Count(abbreviation:pc)
pc=giiin["Purchase ID"].nunique()

#Item Price	(abbreviasion:ip)
ip=giiin["Price"].mean()


#total purchase value by item(tpvi)
tpvi=giiin["Price"].sum()

#item price formating
ip=ip.map("${:,.2f}".format)
tpvi=tpvi.map("${:,.2f}".format)

#Create a summary data frame to hold the results
Most_Popular_Items=pd.DataFrame({"purchase count":pc,
                                     "Item Price":ip,
                                      "Total purchase value":tpvi,     
                                })
#Sort the purchase count in descending order
Most_Popular_Items=Most_Popular_Items.sort_values("purchase count",ascending=False)


Most_Popular_Items

purchase count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   
...                                                              ...   
104     Gladiator's Glaive                                         1   
23      Crucifer                                                   1   
180     Stormcaller                                                1   
91      Celeste                                                    1   
134     Undead Crusader                                            1   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   
...                                                         ...   
104     Gladiator's Glaive                                $1.93   
23      Crucifer                                          $1.99   
180     Stormcaller                                       $3.36   
91      Celeste                                           $4.17   
134     Undead Crusader                                   $4.50   

                                                     Total purchase value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
82      Nirvana                                                    $44.10  
19      Pursuit, Cudgel of Necromancy                               $8.16  
...                                                                   ...  
104     Gladiator's Glaive                                          $1.93  
23      Crucifer                                                    $1.99  
180     Stormcaller                                                 $3.36  
91      Celeste                                                     $4.17  
134     Undead Crusader                                             $4.50  

[183 rows x 3 columns]

## Most Profitable Items

* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [440]:


#total purchase value by item(tpvi)
tpvi=giiin["Price"].sum()


#Create a summary data frame to hold the results
Most_Profitable_Items=pd.DataFrame({"purchase count":pc,
                                    "Item Price":ip,
                                     "Total purchase value":tpvi,     
                               })
#Sort the purchase count in descending order
Most_Profitable_Items=Most_Profitable_Items.sort_values("Total purchase value",ascending=False)
Most_Profitable_Items["Total purchase value"]=Most_Profitable_Items["Total purchase value"].map("${:,.2f}".format)

Most_Profitable_Items

purchase count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
92      Final Critic                                               8   
103     Singed Scalpel                                             8   
...                                                              ...   
125     Whistling Mithril Warblade                                 2   
126     Exiled Mithril Longsword                                   1   
23      Crucifer                                                   1   
104     Gladiator's Glaive                                         1   
42      The Decapitator                                            1   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
92      Final Critic                                      $4.88   
103     Singed Scalpel                                    $4.35   
...                                                         ...   
125     Whistling Mithril Warblade                        $1.00   
126     Exiled Mithril Longsword                          $2.00   
23      Crucifer                                          $1.99   
104     Gladiator's Glaive                                $1.93   
42      The Decapitator                                   $1.75   

                                                     Total purchase value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
92      Final Critic                                               $39.04  
103     Singed Scalpel                                             $34.80  
...                                                                   ...  
125     Whistling Mithril Warblade                                  $2.00  
126     Exiled Mithril Longsword                                    $2.00  
23      Crucifer                                                    $1.99  
104     Gladiator's Glaive                                          $1.93  
42      The Decapitator                                             $1.75  

[183 rows x 3 columns]

# Conclusions
Of the 780 active players, the vast majority are male (84%). There also exists, a smaller, but notable proportion of female layers (14%).

Our peak age demographic falls between 20-24 (46.8%) with secondary groups falling between 15-19 (17.4%) and 25-29 (13%).

The majority of purchases are also done by the age group 20-24 (46.8%) with secondary groups falling between 15-19 (17.4%) and 25-29 (13%).

Out of 183 items offered, the most popular and profitable ones are "Oathbreaker, Last Hope of the Breaking Storm" (12 buys), brought $51 and "Nirvana" and "Fiery Glass Crusader" having (9 buys) each and brought $44 and $41 respectively. Generally, all players (780) prefer different items, there are no significantly more popular item(s) than others.

Average purchase is about $3 per person with the top spenders paying up to $19 for their purchases. Still, 97% are paying way under $10. The total profit from the sold items is about $2400 for 780 players.